### PR TITLE
Addressing Node 18 todos

### PR DESCRIPTION
**Pull request by mrickard**
## Proposed Release Notes

 * Updated the minimum version of the newrelic agent peer dependency to be `>=8.7.0`.

## Links

 * Closes https://issues.newrelic.com/browse/NR-35252
 

## Notes

Removed agent version check as per TODOs
Removed unused semver dependency
Addressed linter warning 
Bumped version

<sup> Created from JetBrains using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

**Comment by michaelgoin** [comment]
I think what is implied with that comment is that we'd update the peer dependency field to require the minimum version of the agent as a breaking (semver major) change along with the add Node 18 / Drop Node 12 work. That is what allows us to safely drop the semver check from this module.

@bizob2828 does that align with your memory?

As such, I think you'll want to update the peer dependency field for `newrelic` in package.json to be >= 8.7.0 and we can slap a semver: major on label on this PR. I don't think we need to hold off merging as I don't expect any changes to come in prior to starting the major release process.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@newrelic/aws-sdk",
-  "version": "4.1.2",
```
Comment (bizob2828) [inline package.json:3]: this is something we do as part of releasing and should not be touched in a different PR

**Comment by bizob2828** [comment]
> 

Yea I was just thinking through this and I think this is correct.

File: tests/versioned/v3/client-dynamodb.tap.js [hunk 2]
```diff
@@ -184,21 +184,17 @@ tap.test('DynamoDB', (t) => {
     ]
   }
 
-  function finish(t, commands, tx) {
```
Comment (bizob2828) [inline tests/versioned/v3/client-dynamodb.tap.js:187]: was there a reason for this change. it's fine but doesn't seem necesssary

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mrickard) [reply package.json:3]: Ah, ok. I wasn't sure if this was handled in github hooks, or if we had to bump versions manually.

File: tests/versioned/v3/client-dynamodb.tap.js (hunk 2 above)
Comment (mrickard) [reply tests/versioned/v3/client-dynamodb.tap.js:187]: I was getting linter complaints about `commands` being shadowed in this inner scope, so I changed it to remove the warning.

File: package.json (hunk 1 above)
Comment (mrickard) [reply package.json:3]: I'll restore the version.

Comment (bizob2828) [reply package.json:3]: Yea so when we get ready to release this. we will prepare release by running https://github.com/newrelic/node-newrelic-aws-sdk/actions/workflows/prepare-release.yml, and picking `major` as the version which will open a PR with updates to changelog and properly update the package.json and package-lock with the new version

Comment (mrickard) [reply package.json]: I've restored the previous version, so this should be re-reviewable.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**